### ISSUE TITLE
Refactor and maintain A22 database tables / indexes

### ISSUE THREAD

**Issue by clezag** [noi-techpark/a22-traffic-connector#10]
Database has steadily grown, to the point where we now have a 300+ GB traffic table, with multiple indexes of 100+ GB.

CPU load is high, correlating with busy day time and falling during nighttime. 

There is already a similar issue open for mobility data: https://github.com/noi-techpark/bdp-core/issues/270

We should try a few things to reign in the exploding database:
- determine if we can delete old records or shift them to an archive table
- partitioning
- rebuild and adjust the already optimized partial index for timestamps > dec 2021
- Remove the btree stationcode index that overlaps with the partial one
- try brin indexes for timestamps
- optimize column data types as most of them are needlessly large

**Comment by clezag** [comment]
During the last modifications to the a22 traffic connector, the full stationcode index was removed, and the partial index updated to oct 2024, freeing up a few 100GB of space, and lowering CPU load

**Comment by sseppi** [comment]
@clezag considering the optimization you did in the first half og 2026, is this issue still valid or can we close it?

**Comment by clezag** [comment]
@sseppi no, this is about the A22 database. We will probably do this the next time we run into low storage or do major changes to that pipeline
